### ISSUE TITLE
Locomotives & tank minecart can get stuck in blocks

### ISSUE THREAD

**Issue by m-nelsen** [railcraft-reborn/railcraft#51]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Bug

 - When a locomotive or a "Minecart with Tank" is placed on a rail, it isn't centered on the rail. If there is a block next to a rail and a railcraft cart/locomotive is placed on it, it'll be partially stuck in the block. Nudging it will get it completely stuck (see screenshots below).

### To Reproduce

 - Place any block down and a rail next to it
 - Place a locomotive or a "Minecart with Tank" on the track
 - You'll see it partially stuck in the block
 - Nudge the cart/locomotive into the block
 - You'll see it become completely stuck (and darkened)

### Expected behavior

 - Railcraft carts and locomotives should work like vanilla minecarts do. When placed down, they should be centered on the rail

### Logs & Environment

```shell
OS: Ubuntu 20.04.5 LTS
Forge 1.19.2-43.1.30

Mods Installed:
 - jei-1.19.2-forge-11.3.0.262
 - railcraft-1.19.2-13.0.0-BETA-1-CI-1ebd203
```

### Stuck cart
![image](https://user-images.githubusercontent.com/49181107/192886391-e950d565-8f5d-405e-a77d-44c09675381a.png)
![image](https://user-images.githubusercontent.com/49181107/192888814-93b5be14-2fd4-4a78-9194-d33ae90cbb1a.png)

### Vanilla minecarts
![image](https://user-images.githubusercontent.com/49181107/192889671-39367fc6-4c0d-4436-bdfd-8d2c478a9fb9.png)

**Comment by Edivad99** [comment]
This should be closed in #59

**Comment by josephtarbit** [comment]
Fixed in [65ddb84](https://github.com/Sm0keySa1m0n/Railcraft/commit/65ddb846bedb3cf238e7ddb39884a673f21c003e)
